app/main.py

```diff
@@ -32,13 +44,15 @@ def main() -> None:
     ]
 
     # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await asyncio.gather(
+        service.run_program(wake_up_program),
+        service.run_program(sleep_program)
+    )
```

Comment:
The current implementation runs both programs in parallel, which might break the logical sequence of operations. Consider using `run_sequence` and `run_parallel` to ensure the correct order of execution for each program.